providers/os/connection/device/device_connection.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package device
+
+import (
+	"errors"
+	"runtime"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/afero"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v11/providers/os/connection/device/linux"
+	"go.mondoo.com/cnquery/v11/providers/os/connection/fs"
+	"go.mondoo.com/cnquery/v11/providers/os/connection/shared"
+	"go.mondoo.com/cnquery/v11/providers/os/id"
+	"go.mondoo.com/cnquery/v11/providers/os/id/ids"
+)
+
+const PlatformIdInject = "inject-platform-ids"
+
+type DeviceConnection struct {
+	*fs.FileSystemConnection
+	plugin.Connection
+	asset         *inventory.Asset
+	deviceManager DeviceManager
+}
+
+func getDeviceManager(conf *inventory.Config) (DeviceManager, error) {
+	shell := []string{"sh", "-c"}
+	if runtime.GOOS == "darwin" {
+		return nil, errors.New("device manager not implemented for darwin")
+	}
+	if runtime.GOOS == "windows" {
+		// shell = []string{"powershell", "-c"}
+		return nil, errors.New("device manager not implemented for windows")
+	}
+	return linux.NewLinuxDeviceManager(shell, conf.Options)
+}
+
+func NewDeviceConnection(connId uint32, conf *inventory.Config, asset *inventory.Asset) (*DeviceConnection, error) {
+	manager, err := getDeviceManager(conf)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug().Str("manager", manager.Name()).Msg("device manager created")
+
+	mi, err := manager.IdentifyBlock(conf.Options)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug().Str("device", mi.DeviceName).Msg("identified block for mounting")
+
+	res := &DeviceConnection{
+		Connection:    plugin.NewConnection(connId, asset),
+		deviceManager: manager,
+		asset:         asset,
+	}
+
+	scanDir, err := manager.Mount()
```

Comment:
I think we should pass in the device we want to mount to make this more re-usable. We can also do this later when we add discovery for multiple block devices.

Reply:
great idea, adjusted. i will clean up the actual volume mounter too in another PR-